cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -87,24 +82,22 @@
   private final DatasetFramework datasetFramework;
   private final MultiThreadDatasetCache multiThreadDatasetCache;
   private final CConfiguration cConf;
-  private final Store store;
   private ListeningExecutorService taskExecutorService;
   private volatile boolean stopping = false;
 
 
   @Inject
   NotificationSubscriberService(MessagingService messagingService,
-                                CConfiguration cConf, Store store,
+                                CConfiguration cConf,
                                 DatasetFramework datasetFramework,
                                 TransactionSystemClient txClient) {
     this.cConf = cConf;
-    this.store = store;
     this.messagingContext = new MultiThreadMessagingContext(messagingService);
     this.multiThreadDatasetCache = new MultiThreadDatasetCache(
       new SystemDatasetInstantiator(datasetFramework), txClient,
       NamespaceId.SYSTEM, ImmutableMap.<String, String>of(), null, null, messagingContext);
     this.transactional = Transactions.createTransactionalWithRetry(
-      Transactions.createTransactional(multiThreadDatasetCache),
+      Transactions.createTransactional(multiThreadDatasetCache, Schedulers.SUBSCRIBER_TX_TIMEOUT_SECONDS),
```

Comment:
I introduced this make sure that the subscriber and constraint checker use the same value. Otherwise, if the subscriber is changed to use a longer timeout, then there may be conflicts. Having this constant ensures that whoever might change it in the future will be aware that the value is coordinated with some other code. Do you think it is an issue to hard code this?